specifyweb/frontend/js_src/lib/components/DataModel/__tests__/resourceApi.test.ts

```diff
@@ -199,20 +199,30 @@ describe('rgetCollection', () => {
 });
 
 describe('needsSaved', () => {
-  test('changing field makes needsSaved true', () => {
+  test('changing field makes needsSaved true', async () => {
     const resource = new tables.CollectionObject.Resource({
       id: collectionObjectId,
     });
+
+    // Need to wait until the business rule fetches Collection -> CollectionObjectType (https://github.com/specify/specify7/pull/5224)
+    // See: https://stackoverflow.com/a/51045733
+    await new Promise(process.nextTick);
```

Comment:
The test was failing as `resource.needsSaved` doesn't update until the `rgetPromise` in this business rule resolves.